WarpBuf/src/WarpBufChugin.cpp

```diff
@@ -503,36 +205,38 @@ void WarpBufChugin::tick(SAMPLE* in, SAMPLE* out, int nframes)
 
     int count = -1;
     int numAvailable = m_rbstretcher->available();
+    int allowedReadCount = 0;
     while (numAvailable < nframes) {
 
-        int allowedReadCount = m_clipInfo.loop_on ? std::min(ibs, (int)( loop_end_seconds*sfinfo.samplerate- sfReadPos)) : ibs;
+        if (sfReadPos > -1 && sfReadPos < sfinfo.frames) {
+            allowedReadCount = m_clipInfo.loop_on ? std::min(ibs, (int)(loop_end_seconds * sfinfo.samplerate - sfReadPos)) : ibs;
+            count = sf_readf_float(sndfile, m_interleavedBuffer, allowedReadCount);
+            sfReadPos += count;
+        }
+        else {
+            count = 0;
+        }
 
-        count = sf_readf_float(sndfile, m_interleavedBuffer, allowedReadCount);
-        sfReadPos += count;
         if (count <= 0) {
-            if (!m_clipInfo.loop_on) {
-                // we're not looping, so just fill with zeros.
-                count = ibs;
-                for (size_t c = 0; c < channels; c++) {
-                    for (int i = 0; i < count; i++) {
-                        m_interleavedBuffer[i*channels+c] = 0.;
-                    }
-                }
-            }
-            else {
+            if (m_clipInfo.loop_on) {
                 // we are looping, so seek to the loop start of the audio file
                 setPlayhead(m_clipInfo.loop_start);
+            }
 
-                allowedReadCount = std::min(ibs, (int)(loop_end_seconds * sfinfo.samplerate - sfReadPos));
-                count = sf_readf_float(sndfile, m_interleavedBuffer, allowedReadCount);
-                sfReadPos += count;
+            if (sfReadPos < 0 || sfReadPos >= sfinfo.frames) {
+                count = 1;
+                // fill with zeros
+                for (int c = 0; c < m_channels; c++) {
+                    for (int i = 0; i < count; i++) {
+                        m_interleavedBuffer[i * m_channels + c] = 0.;
+                    }
+                }
             }
         }
-
-        for (size_t c = 0; c < channels; ++c) {
+        
```

Comment:
delete pls

Reply:
Sorry I can't tell from GitHub what you mean.

Reply:
There was some extra whitespace somewhere, but now I can't find it so nevermind I guess